### DISCUSSION TITLE
Can find printer zywell z303 but cannot find xprinter xp-n160ii, both use emulation esc/pos

### DISCUSSION THREAD

**Discussion by Cuong2601** [tr3v3r/react-native-esc-pos-printer#174] [Q&A]
I'm writing an app that connect printers through LAN, the lib can find zywell printer z303 but cannot find xprinter xp-n160ii. I don't know how. I set up ip address and can find them with the app quick printer. Can you please help me, i'm stuck in here :(

React native: 0.71.7

react-native-esc-pos-printer: 3.1.1

**Comment by tr3v3r** [comment]
@Cuong2601  hello!

For me, it is a big surprise that you able to find **zywell** printer since the lib supports only devices produced by EPSON.

Did you manage to print anything?

**Reply by Cuong2601** [reply]
Yes i can print anything, zywell z303 has device name as TM-m30 when i scan it on LAN. I thought the lib can find all printers that have ESC/POS emulation but you said that it only supports EPSON's devices. 
If you know something could you explain for me?
Thank you very much.

**Reply by tr3v3r** [reply]
It seems in has EPSON's firmware on board. Since TM-m30 this is name of EPSON printer. Did you do anything for that? Or was it like this out of the box?

Probably because of this:

<img width="1323" alt="image" src="https://github.com/user-attachments/assets/6f10e00c-d5b9-4589-aae2-5dbcc18db6a1">

**Reply by Cuong2601** [reply]
I just randomly found it when used the lib. So because it has EPSON's firmware, i see. Could you give me some advice how to scan of find devices that have ESC/POS emulation because it work on zywell but not on xprinter? I'm a newbie in RN, thank you.

**Reply by tr3v3r** [reply]
I'm afraid this lib won't support this. You should try to search other libs that may support it

**Reply by Cuong2601** [reply]
Thank you very much, i will try some.
